processor/lsmintervalprocessor/processor_test.go

```diff
@@ -352,19 +357,35 @@ func benchmarkAggregation(b *testing.B, ottlStatements []string) {
 			md, err := golden.ReadMetrics(filepath.Join(dir, "input.yaml"))
 			require.NoError(b, err)
 			md.MarkReadOnly()
-			b.ResetTimer()
-
-			err = mgp.Start(context.Background(), componenttest.NewNopHost())
-			require.NoError(b, err)
-			for i := 0; i < b.N; i++ {
+			benchFunc := func() {
+				b.StopTimer()
```

Comment:
[**For reviewers**] Not a fan of the copying the pmetric and then adding things to resource attribute but I didn't want to make a lot of change in the benchmark and I also didn't have any other ideas.